src/boundaries.jl

```diff
@@ -0,0 +1,37 @@
+abstract type AbstractBoundary end
+abstract type PrimitiveBoundary <: AbstractBoundary end
+struct Periodic <: PrimitiveBoundary end
+struct Open <: PrimitiveBoundary end
+
+# Helical BCs is a fast approximation to
+# Periodic BCs that takes advantage of the linear storage of
+# arrays. Those familiar with DMRG may recognize this as
+# the "snake" boundary condition.
+struct Helical <: PrimitiveBoundary end
+
+# Allows defining lattices with different BCs along each
+#  dimension. Should reject Helical BC as that is a "global" BC.
+struct MixedBoundary{N} <: AbstractBoundary
+    boundaries::NTuple{N, AbstractBoundary}
```

Comment:
```suggestion
    boundaries::Vector{AbstractBoundary}
```

or 

```julia
struct MixedBoundary{N, BT <: Tuple}
    boundaries::BT
end
```

Reply:
gonna skip this since we've decided to store the tuple of BCs in the Lattice struct